### PR TITLE
in resolution of [Wsign-compare] warning ids : [16, 18, 19, 20, 21 ]

### PR DESCRIPTION
**Pull request by tg-at-google**
@mihaimaruseac

### CODE REVIEW

File: tensorflow/core/platform/file_system.cc [hunk 1]
```diff
@@ -309,7 +309,7 @@ StringPiece FileSystem::Extension(StringPiece path) const {
   StringPiece basename = this->Basename(path);
 
   int pos = basename.rfind('.');
```
Comment (mihaimaruseac) [inline tensorflow/core/platform/file_system.cc:311]: I would declare `pos` here to be `size_t`. No need for a `static_cast` either since `rfind` returns `size_type` already

File: tensorflow/core/profiler/internal/parse_annotation.cc [hunk 2]
```diff
@@ -51,7 +51,7 @@ std::vector<absl::string_view> SplitPairs(absl::string_view metadata) {
   std::vector<absl::string_view> key_value_pairs;
   std::stack<char> quotes;
   int start = 0, end = 0;
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/internal/parse_annotation.cc:53]: I would declare `start` and `end` to be `size_t`

File: tensorflow/core/platform/file_system.cc (hunk 1 above)
Comment (kyscg) [inline tensorflow/core/platform/file_system.cc:311]:
```suggestion
  size_t pos = basename.rfind('.');
```

File: tensorflow/core/profiler/internal/parse_annotation.cc (hunk 2 above)
Comment (kyscg) [inline tensorflow/core/profiler/internal/parse_annotation.cc:53]:
```suggestion
  size_t start = 0, end = 0;
```

**Review by kyscg — COMMENTED** [review]
Applying @mihaimaruseac 's suggestions

**Review by tg-at-google — COMMENTED** [review]
@mihaimaruseac

File: tensorflow/core/profiler/internal/parse_annotation.cc [hunk 3]
```diff
@@ -50,8 +50,8 @@ std::vector<absl::string_view> SplitNameAndMetadata(
 std::vector<absl::string_view> SplitPairs(absl::string_view metadata) {
   std::vector<absl::string_view> key_value_pairs;
   std::stack<char> quotes;
-  int start = 0, end = 0;
-  for (; end < metadata.size(); ++end) {
+  size_t start = 0, end = 0;
+  for (; static_cast<size_t>(end) < metadata.size(); ++end) {
```
Comment (mihaimaruseac) [inline tensorflow/core/profiler/internal/parse_annotation.cc:54]: You no longer need the static cast here

**Review by tg-at-google — COMMENTED** [review]
in response of suggestions

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by tg-at-google** [comment]
@mihaimaruseac

File: tensorflow/core/framework/tensor_shape.cc [hunk 4]
```diff
@@ -187,7 +187,7 @@ void TensorShapeBase<Shape>::InitDims(gtl::ArraySlice<int64> dim_sizes) {
                 "bad overflow check");
   bool large_size = false;
   for (auto s : dim_sizes) {
-    if (s > kMaxSmall) {
+    if (static_cast<size_t>(s) > static_cast<size_t>(kMaxSmall)) {
```
Comment (mihaimaruseac) [inline tensorflow/core/framework/tensor_shape.cc:190]: This seems weird. Let's replace kMaxSmall type to int64 instead

File: tensorflow/core/lib/io/inputbuffer.cc [hunk 5]
```diff
@@ -85,7 +85,7 @@ Status InputBuffer::ReadNBytes(int64 bytes_to_read, string* result) {
   result->resize(bytes_to_read);
   size_t bytes_read = 0;
   Status status = ReadNBytes(bytes_to_read, &(*result)[0], &bytes_read);
```
Comment (mihaimaruseac) [inline tensorflow/core/lib/io/inputbuffer.cc:87]: Would be better to replace the signature of ReadNBytes to not have `size_t` as the type of the last argument. Let's try to do this in another PR (as there would be multiple places to change)

**Review by mihaimaruseac — DISMISSED** [review]

File: tensorflow/core/lib/io/inputbuffer.cc (hunk 5 above)
Comment (tg-at-google) [reply tensorflow/core/lib/io/inputbuffer.cc:87]: Ok, what I'll do is to revert all the files that produce warnings related ReadNBytes(), and as you recommend address them in another PR.

**Review by mihaimaruseac — APPROVED** [review]
